dbms/src/Interpreters/Join.cpp

```diff
@@ -930,6 +1090,17 @@ Block Join::doJoinBlockHash(ProbeProcessInfo & probe_process_info) const
     return block;
 }
 
+Block Join::removeUselessColumn(Block & block) const
+{
+    Block projected_block;
```

Comment:
`assert(!block);`?

Reply:
It shouldn't be necessary. This function works well even if the `block` is empty.